src/cryptonotecore/ValidateTransaction.cpp

```diff
@@ -117,6 +117,24 @@ TransactionValidationResult ValidateTransaction::revalidateAfterHeightChange()
         return m_validationResult;
     }
 
+    /* Validate the transaction inputs are non empty, key images are valid, etc. */
+    if (!validateTransactionInputs())
```

Comment:
This works fine but is a bit overkill to revalidate inputs/outputs just to calculate the sum of inputs / sum of outputs. I would prefer manually looping over inputs/outputs & setting the required variables for `validateTransactionFee()` to work.

Reply:
room for improvement for the next time :-) ty!